ambry-frontend/src/main/java/com/github/ambry/frontend/UserQuotaRequestCostPolicy.java

```diff
@@ -112,7 +114,11 @@ private double calculateCapacityUnitCost(RestRequest restRequest, RestResponseCh
         double cost = Math.ceil(contentSize / CU_COST_UNIT);
         return (cost > 0) ? cost : 1;
       case GET:
-        if (restRequest.getUri().contains(Operations.GET_SIGNED_URL)) {
+        String uri = restRequest.getUri();
+        if (uri.contains(Operations.GET_SIGNED_URL)) {
+          return INDEX_ONLY_COST;
```

Comment:
getSignedUrl calls do not even check for blob existence so could they be considered near zero cost?

Also, once `RequestPath` is used, you could do `getRequestPath(restRequest).matchesOperation(GET_SIGNED_URL)`

Reply:
For now we are going with the assumption that any call to Ambry incurs a min INDEX_ONLY_COST. The min cost is same as cost to fetch a 4MB chunk, and Ambry CAPACITY_UNIT is defined in terms of 4MB reads/writes.

Do you think we should change the granularity of that? Is it overcharging to say any request to Ambry incurs cost of 1 CAPACITY_UNIT with the definition above?

Reply:
We could make INDEX_ONLY_COST configurable, and set it to a fractional value (like 0.25) if we wanted to.

Reply:
For now, maybe we can just keep it as whole units. If we see that heavy users of signed URLs appear to be charged too much, maybe we can consider reducing the cost of the operations that don't even hit the storage nodes.